esinstall/src/rollup-plugins/rollup-plugin-wrap-install-targets.ts

```diff
@@ -98,6 +99,13 @@ export function rollupPluginWrapInstallTargets(
             'const exports={}; const module={exports}; ' + fileContents + ';; module.exports;',
           ),
         );
+
+        // Verify that all of these are valid identifiers. Otherwise when we attempt to
+        // reexport it will produce invalid js like `import { a, b, 0, ) } from 'foo';
+        const allValid = exports.every((identifier) => isValidIdentifier(identifier));
+        if (!allValid) {
```

Comment:
Purely out of curiosity, why return nothing if there was a bad import instead of just `filter()`ing out the bad ones? Is it reasonable to assume that if one is bad, they all are, and would only cause issues by leaving? I think it’s better to be safe and not pollute the namespace if we can help it, but wanted to double-check if you were thinking the same way.

Reply:
+1 for filtering out the bad ones instead

Reply:
I’m not sure I’m suggesting filtering; I was more trying to think out loud about the tradeoffs between the two approaches.

Maybe it’s just an upstream issue with part of the CJS conversion that we should investigate more.

Reply:
I thought a little about it but wasn't sure what would happen if you omit some properties, would they be mistakenly tree-shaken away? Thinking about it more I think they would *not*, so probably filtering is ok. I'd want to add a build test to be sure though.